src/globalshortcuts.c

```diff
@@ -0,0 +1,556 @@
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authors:
+ *       Aleix Pol Gonzalez <aleixpol@kde.org>
+ */
+
+#include "config.h"
+
+#include <string.h>
+#include <glib-object.h>
+
+#include "globalshortcuts.h"
+#include "request.h"
+#include "session.h"
+#include "permissions.h"
+#include "xdp-dbus.h"
+#include "xdp-impl-dbus.h"
+#include "xdp-utils.h"
+
+typedef struct _GlobalShortcuts GlobalShortcuts;
+typedef struct _GlobalShortcutsClass GlobalShortcutsClass;
+
+static GQuark quark_request_session;
+
+struct _GlobalShortcuts
+{
+  XdpDbusGlobalShortcutsSkeleton parent_instance;
+};
+
+struct _GlobalShortcutsClass
+{
+  XdpDbusGlobalShortcutsSkeletonClass parent_class;
+};
+
+static XdpDbusImplGlobalShortcuts *impl;
+static GlobalShortcuts *global_shortcuts;
+
+GType global_shortcuts_get_type (void) G_GNUC_CONST;
+static void global_shortcuts_iface_init (XdpDbusGlobalShortcutsIface *iface);
+
+G_DEFINE_TYPE_WITH_CODE (GlobalShortcuts, global_shortcuts, XDP_DBUS_TYPE_GLOBAL_SHORTCUTS_SKELETON,
+                         G_IMPLEMENT_INTERFACE (XDP_DBUS_TYPE_GLOBAL_SHORTCUTS, global_shortcuts_iface_init));
+
+typedef struct _GlobalShortcutsSession
+{
+  Session parent;
+
+  gboolean closed;
+} GlobalShortcutsSession;
+
+typedef struct _GlobalShortcutsSessionClass
+{
+  SessionClass parent_class;
+} GlobalShortcutsSessionClass;
+
+GType global_shortcuts_session_get_type (void);
+
+G_DEFINE_TYPE (GlobalShortcutsSession, global_shortcuts_session, session_get_type ())
+
+static void
+global_shortcuts_session_close (Session *session)
+{
+  GlobalShortcutsSession *global_shortcuts_session = (GlobalShortcutsSession *)session;
+
+  global_shortcuts_session->closed = TRUE;
+}
+
+static void
+global_shortcuts_session_finalize (GObject *object)
+{
+  G_OBJECT_CLASS (global_shortcuts_session_parent_class)->finalize (object);
+}
+
+static void
+global_shortcuts_session_init (GlobalShortcutsSession *global_shortcuts_session)
+{
+}
+
+static void
+global_shortcuts_session_class_init (GlobalShortcutsSessionClass *klass)
+{
+  GObjectClass *object_class;
+  SessionClass *session_class;
+
+  object_class = G_OBJECT_CLASS (klass);
+  object_class->finalize = global_shortcuts_session_finalize;
+
+  session_class = (SessionClass *)klass;
+  session_class->close = global_shortcuts_session_close;
+}
+
+static GlobalShortcutsSession *
+global_shortcuts_session_new (GVariant *options,
+                         Request *request,
+                         GError **error)
```

Comment:
Please align the function arguments so that they start after the parenthesis column. Do this here, throughout this entire file, and also the header file. It should look like this:

```c
static GlobalShortcutsSession *
global_shortcuts_session_new (GVariant *options,
                              Request *request,
                              GError **error)
```

Reply:
I'll do my best. Is there a clang-format config for it?